src/go/rpk/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/redpanda-data/redpanda/src/go/rpk
 
-go 1.23.1
+go 1.24.3
```

Comment:
I think there are other go.mod that need updates... What do you think about rolling up all the go.mod files into a single one in the project root? That doesn't have to happen in this PR I can do that, but want to get your thoughts.